qiskit/providers/aer/noise/noise_model.py

```diff
@@ -208,7 +213,8 @@ def from_backend(cls, backend,
                      gate_lengths=None,
                      gate_length_units='ns',
                      standard_gates=None,
-                     warnings=True):
+                     warnings=True,
+                     delay_noise=False):
```

Comment:
I agree with the change of the meaning of `delay_noise` option.
However, I have some reservations on changing its default value.
It will drastically change the default behavior of `NoiseModel.from_backend`. All the existing codes using default noise model will stop working and raise errors that tells circuits must be scheduled. That's why I'm hesitating to set `delay_noise=True`. Do you think that is OK?

Reply:
No errors should be raised (if they are that is something that needs to change).

The relaxation noise noise pass should not care if the input is a scheduled circuit or not. It only looks for the specified instructions (delay in this case) and add noise if present with a non-zero duration. Scheduling is an optional step you can do to add delays. If you don't schedule your circuit, your simulation will be exactly the same as before (unless you had manually added delays to your circuit). If you do schedule your circuit then you will get the extra delay noise (this should be added to API documentation for noise model, and basic device noise tutorial)

In general the relaxation noise pass should also work on a non-scheduled circuit by doing nothing (because the instructions would have `None` for duration if not scheduled).

This reminds me that we need to add a release note explaining new additions. In particular it should have both a `feature` section (for the new passes and addition to noise model) and `upgrade` section that points out this change if you run a scheduled circuit or circuit with delays.

Reply:
I understand no errors should be raised for non-scheduled circuit. In 1672eb0, I've updated the relaxation noise pass so that it can add noises to gates as well as delays relying on `InstructionDurations` supplied instead of `dt`. It's different from your suggestion but what do you think of it? does it make sense? It could be seen as a problem who the relaxation noise pass should ask to know durations, `InstructionDurations` or `Instruction.duration`. (Both of them might be deprecated after circuit has durations info in its metadata, though)